src/permission/net_permission.cc

```diff
@@ -0,0 +1,172 @@
+    std::bitset<IPV6_BIT_LEN> network = netmask & ip;
+    net_mask_node = std::make_unique<NetMaskNode>(netmask, network);
+  }
+  auto node = std::make_unique<NetNode>(
+      result->address, result->port, std::move(net_mask_node));
+  granted_udp_.push_back(std::move(node));
+  deny_all_udp_ = false;
+}
+
+bool NetPermission::is_granted(Environment* env,
+                               PermissionScope perm,
+                               const std::string_view& param = "") const {
+  switch (perm) {
+    case PermissionScope::kNetUDP:
+      return !deny_all_udp_ &&
+             (allow_all_udp_ || CheckoutResource(granted_udp_, param));
+    default:
+      return false;
+  }
+}
+
+bool NetPermission::CheckoutResource(
+    const std::vector<std::unique_ptr<NetNode>>& list,
+    const std::string_view& param) const {
+  if (param.empty()) {
+    return false;
+  }
+  using std::string_view_literals::operator""sv;
+  const std::vector<std::string_view> address_port = SplitString(param, "/"sv);
+  int size = address_port.size();
+  CHECK(size == 1 || size == 2);
+  std::string address;
+  std::string port;
+  if (size == 2) {
+    address = address_port[0];
+    port = address_port[1];
+  } else {
+    address = address_port[0];
+    port = "*";
+  }
+  for (uint32_t i = 0; i < list.size(); i++) {
+    auto node = list[i].get();
+    int ip_version = IsIP(address);
+    if (ip_version > 0 && node->net_mask_node) {
+      std::bitset<IPV6_BIT_LEN> ip;
+      if (!ConvertIPToBitset(&ip, address, ip_version)) {
+        return false;
+      }
+      std::bitset<IPV6_BIT_LEN> network = node->net_mask_node->netmask & ip;
+      if (node->net_mask_node->network == network &&
+          (node->port == "*" || node->port == port)) {
+        return true;
+      }
+    } else if ((node->address == "*" || node->address == address) &&
+               (node->port == "*" || node->port == port)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+void NetPermission::Print() const {
+  if (UNLIKELY(per_process::enabled_debug_list.enabled(
+          DebugCategory::PERMISSION_MODEL))) {
+    std::cout << "net-udp-net: " << std::endl;
+    std::cout << "  deny_all_udp_: " << deny_all_udp_ << std::endl;
+    std::cout << "  allow_all_udp_: " << allow_all_udp_ << std::endl;
+    JSONWriter writer(std::cout, false);
+    writer.json_arraystart("Nodes");
+    for (uint32_t i = 0; i < granted_udp_.size(); i++) {
+      auto node = granted_udp_[i].get();
+      writer.json_start();
+      writer.json_keyvalue("address", node->address);
+      writer.json_keyvalue("port", node->port);
+      if (node->net_mask_node) {
+        writer.json_keyvalue("netmask",
+                             node->net_mask_node->netmask.to_string());
+        writer.json_keyvalue("network",
+                             node->net_mask_node->network.to_string());
+      }
+      writer.json_end();
+    }
+    writer.json_arrayend();
+    std::cout << std::endl;
+  }
+}
```

Comment:
```suggestion
```

Reply:
This function only use when `per_process::enabled_debug_list.enabled(DebugCategory::PERMISSION_MODEL))` is true.